File: ios/AdchainSdkPartnerSample/AdchainOfferwallViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(AdchainOfferwallViewManager, RCTViewManager)

// React Native Props
RCT_EXTERN_METHOD(setPlacementId:(UIView *)view placementId:(NSString *)placementId)
RCT_EXTERN_METHOD(setAppKey:(UIView *)view appKey:(NSString *)appKey)
RCT_EXTERN_METHOD(setBaseUrl:(UIView *)view baseUrl:(NSString *)baseUrl)
RCT_EXTERN_METHOD(setUserId:(UIView *)view userId:(NSString *)userId)
RCT_EXTERN_METHOD(setPlatform:(UIView *)view platform:(NSString *)platform)
RCT_EXTERN_METHOD(setUrl:(UIView *)view url:(NSString *)url)

// Event Handlers
RCT_EXTERN_METHOD(onOfferwallOpened:(UIView *)view)
RCT_EXTERN_METHOD(onOfferwallClosed:(UIView *)view)
RCT_EXTERN_METHOD(onOfferwallError:(UIView *)view message:(NSString *)message)
RCT_EXTERN_METHOD(onRewardEarned:(UIView *)view amount:(NSInteger)amount)

@end

File: android/app/src/main/java/com/adchainsdkpartnersample/AdchainOfferwallViewManager.kt
package com.adchainsdkpartnersample

import com.facebook.react.bridge.*
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import android.util.Log

/**
 * AdchainOfferwallView를 React Native에서 사용하기 위한 ViewManager
 */
class AdchainOfferwallViewManager : SimpleViewManager<AdchainOfferwallView>() {
    
    companion object {
        const val NAME = "AdchainOfferwallView"
    }
    
    override fun getName(): String = NAME
    
    override fun createViewInstance(reactContext: ThemedReactContext): AdchainOfferwallView {
        Log.d("AdchainOfferwallViewManager", "Creating AdchainOfferwallView instance")
        return AdchainOfferwallView(reactContext)
    }
    
    @ReactProp(name = "placementId")
    fun setPlacementId(view: AdchainOfferwallView, placementId: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting placementId: $placementId")
        view.placementId = placementId
    }
    
    @ReactProp(name = "appKey")
    fun setAppKey(view: AdchainOfferwallView, appKey: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting appKey: $appKey")
        view.appKey = appKey
    }
    
    @ReactProp(name = "baseUrl")
    fun setBaseUrl(view: AdchainOfferwallView, baseUrl: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting baseUrl: $baseUrl")
        view.baseUrl = baseUrl
    }
    
    @ReactProp(name = "userId")
    fun setUserId(view: AdchainOfferwallView, userId: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting userId: $userId")
        view.userId = userId
    }
    
    @ReactProp(name = "platform")
    fun setPlatform(view: AdchainOfferwallView, platform: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting platform: $platform")
        view.platform = platform
    }
    
    @ReactProp(name = "url")
    fun setUrl(view: AdchainOfferwallView, url: String?) {
        Log.d("AdchainOfferwallViewManager", "Setting URL: $url")
        
        if (!url.isNullOrEmpty()) {
            view.loadOfferwall(url)
        }
    }
    
    override fun onDropViewInstance(view: AdchainOfferwallView) {
        Log.d("AdchainOfferwallViewManager", "Dropping AdchainOfferwallView instance")
        view.cleanup()
        super.onDropViewInstance(view)
    }
}
